gk/fib.c

```diff
@@ -1425,18 +1419,15 @@ check_prefix_locked(struct ip_prefix *prefix,
 }
 
 int
-add_fib_entry(const char *prefix, const char *gt_ip, const char *gw_ip,
+add_fib_entry_numerical(struct ip_prefix *prefix_info,
+	struct ipaddr *gt_addr, struct ipaddr *gw_addr,
 	enum gk_fib_action action, struct gk_config *gk_conf)
 {
 	int ret;
-	struct ip_prefix prefix_info;
 	struct gk_fib *neigh_fib;
-	struct ipaddr gt_addr, gw_addr;
-	struct ipaddr *gt_para = NULL, *gw_para = NULL;
 
-	prefix_info.str = prefix;
-	prefix_info.len = parse_ip_prefix(prefix, &prefix_info.addr);
-	if (prefix_info.len < 0)
+	if (prefix_info == NULL || prefix_info->str == NULL
+			|| prefix_info->len < 0)
```

Comment:
Given that `prefix_info` is a required parameter, we don't need the extra tests `prefix_info == NULL || prefix_info->str == NULL` because if the caller forgets them, the code is going to halt. The extra test would make sense if `add_fib_entry_numerical()` could be called from Lua, but there's no reason to call `add_fib_entry_numerical()` from Lua.